sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -1961,12 +1961,98 @@ class DataSourceV2SQLSuite
-      testNotSupportedV2Command("SHOW CREATE TABLE", t)
-      testNotSupportedV2Command("SHOW CREATE TABLE", s"$t AS SERDE")
+      val e = intercept[AnalysisException] {
+        sql(s"SHOW CREATE TABLE $t AS SERDE")
+      }
+      assert(e.message.contains(s"SHOW CREATE TABLE AS SERDE is not supported for v2 tables."))
+    }
+  }
+
+  test("SPARK-33898: SHOW CREATE TABLE") {
+    val t = "testcat.ns1.ns2.tbl"
+    withTable(t) {
+      spark.sql(  s"""CREATE TABLE $t (
+                     |  a bigint,
+                     |  b bigint,
+                     |  c bigint,
+                     |  `extra col` ARRAY<INT>,
+                     |  `<another>` STRUCT<x: INT, y: ARRAY<BOOLEAN>>
+                     |)
+                     |USING foo
+                     |OPTIONS (
+                     |  from = 0,
+                     |  to = 1)
+                     |COMMENT 'This is a comment'
+                     |TBLPROPERTIES ('prop1' = '1')
+                     |PARTITIONED BY (a)
+                     |LOCATION '/tmp'
+         """.stripMargin)
+      val showDDL = getShowCreateDDL(s"SHOW CREATE TABLE $t")
+      assert(showDDL === Array(
+        "CREATE TABLE testcat.ns1.ns2.tbl (",
+        "`a` BIGINT,",
+        "`b` BIGINT,",
+        "`c` BIGINT,",
+        "`extra col` ARRAY<INT>,",
+        "`<another>` STRUCT<`x`: INT, `y`: ARRAY<BOOLEAN>>)",
+        "USING foo",
+        "OPTIONS(",
+        "'from' = '0',",
+        "'to' = '1')",
+        "PARTITIONED BY (a)",
+        "COMMENT 'This is a comment'",
+        "LOCATION '/tmp'",
+        "TBLPROPERTIES(",
+        "'prop1' = '1')"
+      ))
+    }
+  }
+
+  test("SPARK-33898: SHOW CREATE TABLE WITH AS SELECT") {
+    val t = "testcat.ns1.ns2.tbl"
+    withTable(t) {
+      spark.sql(  s"""CREATE TABLE $t
+                     |USING foo
+                     |AS SELECT 1 AS a, "foo" AS b
+         """.stripMargin)
+      val showDDL = getShowCreateDDL(s"SHOW CREATE TABLE $t")
+      assert(showDDL === Array(
+        "CREATE TABLE testcat.ns1.ns2.tbl (",
+        "`a` INT,",
+        "`b` STRING)",
+        "USING foo"
+      ))
+    }
+  }
+
+  test("SPARK-33898: SHOW CREATE TABLE PARTITIONED BY Transforms") {
+    val t = "testcat.ns1.ns2.tbl"
+    withTable(t) {
+      val createSql =
+        s"""
+          |CREATE TABLE $t (a INT, b STRING, ts TIMESTAMP) USING foo
+          |PARTITIONED BY (
+          |    a,
+          |    bucket(16, b),
+          |    years(ts),
+          |    months(ts),
+          |    days(ts),
+          |    hours(ts))
+      """.stripMargin
+      spark.sql(createSql)
+      val showDDL = getShowCreateDDL(s"SHOW CREATE TABLE $t")
+      assert(showDDL === Array(
+        "CREATE TABLE testcat.ns1.ns2.tbl (",
+        "`a` INT,",
+        "`b` STRING,",
+        "`ts` TIMESTAMP)",
+        "USING foo",
+        "PARTITIONED BY (a, bucket(16, b), years(ts), months(ts), days(ts), hours(ts))"
```

Comment:
shall these attributes be quoted too?

Reply:
I think we should not quote the column names in other places instead, if there is no special char.

Reply:
> I think we should not quote the column names in other places instead, if there is no special char.

+1

Reply:
@Peng-Lei can you address this?

> I think we should not quote the column names in other places instead, if there is no special char.